torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,304 @@
+    )
+    zipfile_glob = "nlcd_*_land_cover_l48_20210604.zip"
+    date_format = "%Y"
+    is_image = False
+
+    url = "https://s3-us-west-2.amazonaws.com/mrlc/nlcd_{}_land_cover_l48_20210604.zip"
+
+    md5s = {
+        2001: "538166a4d783204764e3df3b221fc4cd",
+        2006: "67454e7874a00294adb9442374d0c309",
+        2011: "ea524c835d173658eeb6fa3c8e6b917b",
+        2016: "452726f6e3bd3f70d8ca2476723d238a",
+        2019: "82851c3f8105763b01c83b4a9e6f3961",
+    }
+
+    ordinal_label_map = {
+        0: 0,
+        11: 1,
+        12: 2,
+        21: 3,
+        22: 4,
+        23: 5,
+        24: 6,
+        31: 7,
+        41: 8,
+        42: 9,
+        43: 10,
+        52: 11,
+        71: 12,
+        81: 13,
+        82: 14,
+        90: 15,
+        95: 16,
+    }
+
+    cmap = {
+        0: (0, 0, 0, 255),
+        1: (70, 107, 159, 255),
+        2: (209, 222, 248, 255),
+        3: (222, 197, 197, 255),
+        4: (217, 146, 130, 255),
+        5: (235, 0, 0, 255),
+        6: (171, 0, 0, 255),
+        7: (179, 172, 159, 255),
+        8: (104, 171, 95, 255),
+        9: (28, 95, 44, 255),
+        10: (181, 197, 143, 255),
+        11: (204, 184, 121, 255),
+        12: (223, 223, 194, 255),
+        13: (220, 217, 57, 255),
+        14: (171, 108, 40, 255),
+        15: (184, 217, 235, 255),
+        16: (108, 159, 184, 255),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        years: List[int] = [2019],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            years: list of years for which to use nlcd layer
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+            AssertionError: if ``year`` is invalid
+        """
+        assert set(years).issubset(
+            self.md5s.keys()
+        ), f"NLCD data product only exists for the following years: {self.md5s.keys()}."
```

Comment:
```suggestion
        ), f"NLCD data product only exists for the following years: {list(self.md5s.keys())}."
```
Otherwise it looks like:
```
dict_keys([2001, 2006, 2011, 2016, 2019])
```
Might need to wrap onto multiple lines.